core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelector.java

```diff
@@ -168,47 +158,36 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         queueSizeAdjuster.update(existingTasks, assignmentStats);
         Set<InternalNode> blockedExactNodes = new HashSet<>();
         boolean splitWaitingForAnyNode = false;
-        // splitsToBeRedistributed becomes true only when splits go through locality-based assignment
-        boolean splitsToBeRedistributed = false;
-        Set<Split> remainingSplits = new HashSet<>(splits.size());
 
         List<InternalNode> filteredNodes = filterNodes(nodeMap, includeCoordinator, ImmutableSet.of());
         ResettableRandomizedIterator<InternalNode> randomCandidates = new ResettableRandomizedIterator<>(filteredNodes);
         Set<InternalNode> schedulableNodes = new HashSet<>(filteredNodes);
 
-        // optimizedLocalScheduling enables prioritized assignment of splits to local nodes when splits contain locality information
-        if (optimizedLocalScheduling) {
```

Comment:
Can we remove `optimizedLocalScheduling` from NodeSchedulerConfig ?

Reply:
Can you make a separate PR out of the last commit ? It's unrelated to everything else and can be landed quickly

Reply:
I think we can also need to change implementation of `hive.force-local-scheduling`. With current changes we're effectively forcing local scheduling by default in hive (unless node went down). That's probably not the right default as it could create hot spots in the cluster. Instead of flipping `isRemotelyAccessible`, that flag should now change whether or not host addresses are returned by hive connector and by default we shouldn't provide addresses from HDFS.